a3p-integration/proposals/n:upgrade-next/test-lib/provision-helpers.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-env node */
+import { retryUntilCondition } from '@agoric/client-utils';
+import {
+  addUser,
+  CHAINID,
+  makeAgd,
+  VALIDATORADDR,
+} from '@agoric/synthetic-chain';
+import { execFileSync } from 'node:child_process';
+
+const agd = makeAgd({ execFileSync }).withOpts({ keyringBackend: 'test' });
+
+/**
+ * @param {string} addr
+ * @param {string} wanted
+ * @param {string} [from]
+ */
+export const bankSend = (addr, wanted, from = VALIDATORADDR) => {
+  return agd.tx(['bank', 'send', from, addr, wanted], {
```

Comment:
The call to `makeAgd` is just right, except that it's at module scope.
So the exported `bankSend` closes over `execFileSync`.

I'd really rather we stuck to [OCap Discipline](https://github.com/Agoric/agoric-sdk/wiki/OCap-Discipline) house style rather than digging deeper...

 - #2160

Reply:
This is just restoring the relevant a3p tests as they existed in u19, and slightly modifies them as some tests depended on each other. I am not willing to rewrite / refactor them